app/src/core/MainClient.ts

```diff
@@ -50,116 +50,33 @@ export default class MainClient extends EventEmitter {
         this._request = new Request(this._catchErrors);
 
         this._profilesController = new ProfilesController(this._request, this._dispatch);
+        this._mainApi = new MainApi(this._request, this._socket, this._dispatch, this._profilesController);
 
         this._bindProfileListeners();
         this._bindCatchErrorsListeners();
     }
 
-    public catchErrors(error: string) {
+    get mainApi() {
```

Comment:
so as I understand if we make it here and then call mainClient.mainApi, we make this api layer for some dicomposition, right?

Reply:
yes, we can`t keep this api methods in "MainClient" because it is not a main methods. So we should create the new layer named MainApi, which should keep several main methods, like singIn, logout etc